conan/tools/env/environment.py

```diff
@@ -58,18 +58,18 @@ def environment_wrap_command(conanfile, env_filenames, env_folder, cmd, subsyste
     if bats:
         launchers = " && ".join('"{}"'.format(b) for b in bats)
         if ps1s:
-            ps1_launchers = f'{powershell} -Command "' + " ; ".join('&\'{}\''.format(f) for f in ps1s) + '"'
-            cmd = cmd.replace('"', "'")
-            return '{} && {} ; cmd /c {}'.format(launchers, ps1_launchers, cmd)
+            ps1_launchers = f'{powershell} -Command "' + " ; ".join('&\'{}\''.format(f) for f in ps1s)
+            cmd = cmd.replace('"', r'\"')
+            return '{} && {} ; cmd /c {}"'.format(launchers, ps1_launchers, cmd)
         else:
             return '{} && {}'.format(launchers, cmd)
     elif shs:
         launchers = " && ".join('. "{}"'.format(f) for f in shs)
         return '{} && {}'.format(launchers, cmd)
     elif ps1s:
-        ps1_launchers = f'{powershell} -Command "' + " ; ".join('&\'{}\''.format(f) for f in ps1s) + '"'
-        cmd = cmd.replace('"', "'")
-        return '{} ; cmd /c {}'.format(ps1_launchers, cmd)
+        ps1_launchers = f'{powershell} -Command "' + " ; ".join('&\'{}\''.format(f) for f in ps1s)
+        cmd = cmd.replace('"', r'\"')
+        return '{} ; cmd /c {}"'.format(ps1_launchers, cmd)
```

Comment:
Same comment as above
```suggestion
        return '{} ; cmd /c "{}"'.format(ps1_launchers, cmd)
```

Reply:
From what I understood, we first had to remove the trailing `"` after the environment activation in order to put the **whole** powershell command between double quotes. Hence the single one we have here